### ISSUE TITLE
Incorrect and Unclear Summarization Response

### ISSUE THREAD

**Issue by sharkey97** [rojagtap/abstractive_summarizer#7]
After using the code myself and training it on my own computer (which took roughly 30 hours), I'm getting very odd responses from the summarize() function. Using the example provided in your code, I receive the following output:

"infosys clears deal to buy back firm for ₹3 300 cr <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop>"

I've tried other test paragraphs and receive nothing close to what is expected. Unfortunately I'm unable to attach my checkpoints. The only thing I've changed in the code is adding "tf.enable_eager_execution" and changing the print of the "if batch % 429 == 0:" statement to "print(f'Batch: {epoch +1}, epoch: {batch}, Loss: {train_loss.result():.4f}')" Due to compatibility issues.
Any help would be appreciated.

**Comment by rojagtap** [comment]
Hey,

Could you help me with the number of epochs you trained the model for?

**Comment by sharkey97** [comment]
20 Epochs just like the code here.

**Comment by rojagtap** [comment]
It's difficult to say what exactly went wrong, but by the looks of it, I think it is because of high loss. Can you try and train a bit longer, maybe up to something close to 2?

**Comment by sharkey97** [comment]
I can try that yes, although can you suggest a way I could possibly train this quicker? Or carry on training from my previous checkpoint? My last attempt took 30 hours. Sorry if these questions aren't great, I'm quite new to Tensorflow

**Comment by rojagtap** [comment]
No, they're absolutely valid questions :)

I think you can speed up training by using maybe better hardware (better GPU or a TPU maybe).
And regarding restoring the checkpoints, you can refer to cell #13 in my notebook.

You can find more on restoring checkpoints from my article: https://towardsdatascience.com/10-tensorflow-tricks-every-ml-practitioner-must-know-96b860e53c1 (point 8)

**Comment by sharkey97** [comment]
Thanks for you help. 
I'm currently trying to install the correct cuda libraries as it seems the program was running on my CPU and isn't detecting my GPU at all, which must be why it is taking so long. I'll read into what you have suggested and get back to you.

**Comment by rojagtap** [comment]
Why don't you try google colab?

**Comment by sharkey97** [comment]
I've managed to complete another 20 epochs of training, and have also got CUDA working so it's much quicker. My final loss was 1.9707, however my output still doesn't seem right:

infosys approves overseas investment of ₹8 000 cr <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop><stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop> <stop>

**Comment by sharkey97** [comment]
Sorry for the double comment. I have trained another 20 Epochs and now have a loss of 1.6157. But Still not getting the expected results. My output is as follows:

"successful firms form to acquire funding journalists"

**Comment by rojagtap** [comment]
That's strange. Can you confirm using a validation set if your model is overfitting?
